### ISSUE TITLE
Add support for async await

### ISSUE THREAD

**Issue by bordunosp** [grpc/grpc-swift#1205]
are you planning to move to the "async-await" paradigm?
mb there is already some branch for tests?

**Comment by glbrntt** [comment]
Yes, we plan on adopting async await. We don't have anything for users to play around with yet though.

**Comment by bordunosp** [comment]
swift-nio has already demo sources for support swift 5.5
https://github.com/apple/swift-nio/tree/main/Sources/NIOAsyncAwaitDemo

and around this vapor has branch https://github.com/vapor/vapor/tree/async-await

**Comment by glbrntt** [comment]
We now have an experimental release with initial support async/await https://github.com/grpc/grpc-swift/releases/tag/1.4.1-async-await.1

**Comment by Jerry-Carter** [comment]
Thank you for the 1.4.1-async-await branch.  Curious what the roadmap is.  Specifically, are there plans for a 1.5.0-async-await branch?  Do you anticipate this being folded in before 2.0?  Thanks.

**Comment by glbrntt** [comment]
Support for async await was added in the 1.8.0 release.

**Comment by cristian064** [comment]
> Support for async await was added in the 1.8.0 release.

Hi do you know where can I find the documentation about how to use async await on GRPC?

**Comment by glbrntt** [comment]
This is probably the best place to start: https://github.com/grpc/grpc-swift/blob/main/docs/basic-tutorial.md
